src/logout.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { updateInitialState } from './reducer-handlers';
+
+let LogoutDumb = ({ userObject, updateInitialState, history }) => 
+    <li>
+        {
+            shouldRender(userObject, updateInitialState, history)
+        }
+    </li>
+
+let shouldRender = (userObject, updateInitialState, history) => { 
+    if (userObject) {
+        return <h2 onClick={() =>
+            logoutFunction(updateInitialState, history)
+        }
+        >Logout</h2>
+    }
+}
+
+let logoutFunction = (updateInitialState, history) => {
+    localStorage.removeItem('token');
+    updateInitialState('');
+    history.push();
```

Comment:
does not work. prop of history does not appear to be successfully passed down.  Doesn't matter the input for a path.